src/scripts/allocation_receipts.py

```diff
@@ -0,0 +1,199 @@
+               f"GNO Holder Details\n  {holder_details}\n" \
+               f"------------------------------------------\n" \
+               f"Trader Details\n  {trader_details.strip()}\n" \
+               f"------------------------------------------\n" \
+               f"Community Details\n  {poap_details}"
+
+
+def load_all_data_from_out(allocation_files: AllocationFiles) -> AllocationDetails:
+    # Load Holder Info
+    holder_details = HolderDetails(
+        data=index_by_account(
+            CombinedGnoHolder.load_from(
+                load_from=allocation_files.holder_data.combined
+            )
+        ),
+        allocations=IndexedAllocations.load_from_file(
+            file=allocation_files.holder_allocation
+        )
+    )
+
+    # Load Trader Info
+    raw_trader_data = CowSwapTrader.load_and_merge_network_trader_data(
+        mainnet_file=allocation_files.trader_data.traders.filename('mainnet'),
+        gchain_file=allocation_files.trader_data.traders.filename('gchain'),
+    )
+    primary_trader_details = TraderDetails(
+        data=CowSwapTrader.load_from(
+            load_file=allocation_files.trader_data.primary_trader
+        ),
+        allocations=IndexedAllocations.load_from_file(
+            file=allocation_files.trader_data.primary_allocation
+        ),
+        allocation_type=AllocationType.PRIMARY
+    )
+    consolation_details = TraderDetails(
+        data=CowSwapTrader.load_from(
+            load_file=allocation_files.trader_data.consolation_trader
+        ),
+        allocations=IndexedAllocations.load_from_file(
+            file=allocation_files.trader_data.consolation_allocation
+        ),
+        allocation_type=AllocationType.CONSOLATION
+    )
+
+    # Load POAP Info
+    poap_details = PoapDetails(
+        # TODO - load gchain alphas
+        mainnet=Account.load_from(
+            load_file=allocation_files.alpha_traders.filename('mainnet')
+        ),
+        gchain=Account.load_from(
+            load_file=allocation_files.alpha_traders.filename('gchain')
+        ),
+        allocations=IndexedAllocations.load_from_file(
+            file=allocation_files.poap_allocations
+        )
+    )
+
+    return AllocationDetails(
+        holder=holder_details,
+        primary=primary_trader_details,
+        consolation=consolation_details,
+        poap=poap_details,
+        raw_trader_data=raw_trader_data
+    )
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Fetch Allocation Receipts for a list of accounts"
+    )
+    parser.add_argument(
+        "--account-file",
+        type=str,
+        help="file containing ethereum addresses",
+    )
+    args = parser.parse_args()
+    # Load all Data related to allocations and where they were derived from
+
+    allocation_details = load_all_data_from_out(AllocationFiles())
+
+    accounts = sorted(
+        Account.load_from(File(name=args.account_file, path='./out/')),
```

Comment:
I found it counterintuitive that this file is expected in a certain path. If it's passed in as a commandline argument, I'd expect it to be relative to the folder from which I execute the command.

Reply:
In that case the user is going to have to provide both the path and the file name as a runtime argument. I can probably split the full path for them on the last / and deal with it internally.